prov/util/src/util_wait.c

```diff
@@ -142,6 +142,68 @@ int fi_wait_init(struct util_fabric *fabric, struct fi_wait_attr *attr,
 	return 0;
 }
 
+int ofi_wait_fd_match(struct dlist_entry *item, const void *arg)
+{
+	struct ofi_wait_fd_entry *fd_entry;
+
+	fd_entry = container_of(item, struct ofi_wait_fd_entry, entry);
+	return fd_entry->fd == *(int *)arg;
+}
+
+int ofi_wait_fd_del(struct util_wait *wait, int fd)
+{
+	struct ofi_wait_fd_entry *fd_entry;
+	struct dlist_entry *entry;
+	struct util_wait_fd *wait_fd = container_of(wait, struct util_wait_fd,
+						    util_wait);
+
+	fastlock_acquire(&wait_fd->lock);
+	entry = dlist_remove_first_match(&wait_fd->fd_list, ofi_wait_fd_match,
+					 &fd);
+	fastlock_release(&wait_fd->lock);
+	if (!entry) {
+		FI_WARN(wait->prov, FI_LOG_FABRIC,
+			"Given fd not found in wait list\n");
+		return -FI_EINVAL;
+	}
```

Comment:
I think an fd might be added to the same wait set twice, so I would convert the FI_WARN into FI_INFO